File: examples-node/example02.js
var edge = require('@agracio/edge-js/lib/edge');

var sayHello = edge.func('cs',function () {
/*
	async (dynamic input) => 
	{ 
		return String.Format("Hello {0} {1}", input.firstName, input.lastName); 
	}
*/
});

let firstName=process.argv[2];
let lastName=process.argv[3];

sayHello(
	{
		firstName: firstName,
		lastName: lastName
	}, 
	function (error, result) {
		if (error) throw error;
		console.log(result);
	}
);

File: examples-cs/examples-cs/examples-cs/example_02.cs
using EdgeJs;
using System;
using System.Threading.Tasks;

internal class example_02
{
    internal static void exec()
    {
        string javascript = @"
            return function (input, callback) {
                try {
                    const message=`Hello ${input.firstName} ${input.lastName}`;
                    console.log(message);
                    callback(null, `result: ${message}`);
                } catch (error) {
                    callback(error, `result: ${message}`);
                }                
            }
        ";
        Func<Object, Task<Object>> func = Edge.Func(javascript);

        var input = new { firstName = "Ken", lastName = "Hill" };
        string result = func(input).Result as String;
        Console.WriteLine(result);

        //wait for keyboard
        Console.ReadLine();
    }
}
